### ISSUE TITLE
Which genotypers is GLIMPSE compatible with?

### ISSUE THREAD

**Issue by ivargr** [odelaneau/GLIMPSE#96]
You say in the documentation that all GLIMPSE requires in order to do imputation is a VCF where genotype likelyhoods are specified at all variant sites. However, do you have a more specific list of genotypers that you actually have successfully used GLIMPSE with (except for bcftools mpileup)?

I've tried using GLIMPSE with some genotypers (e.g. Malva, Graphtyper, GATK Haplotypecaller), but some of the ones I've tried seem incompatible since they don't output variants that are called as homozygous ref (0/0). It also seems that it varies a bit what "resolution" genotypers used for genotype likelyhoods (some cap the likelyhood at 255, which I suspect may lead to worse results if all phred probabilities are either 0 or 255, since GLIMPSE cannot differ between two variants with 255). 

Anyway, it would be very nice to know if the recommendation is simply to use _bcftools mpileup_ for now with GLIMPSE (like you do in the tutorial), or whether you have successfully used any other genotypers with GLIMPSE.

I also have another question about the required format for genotype likelyhoods, but posting that as a separate issue since it's another question.

Thanks!

**Comment by srubinacci** [comment]
Hi,
Yes, typically I force call genotypes at all known positions of the reference panel (e.g. using the bcftools command I provide, but you can use GATK or freebayes to do the same). You just need to "force-call" all positions provided. Regarding the resolution, I do not really get your point: keep in mind that genotype likelihoods encoded in the PL fields are phread-scaled: this means that having a PL for a genotype equal to 255 or 600 does not change much, they both have practically a value of zero, therefore you can encode each of them as a char. 

I have used many genotype callers. Other than bcftools (as I provide the command in the tutorial), you can for example use freebayes using: 
`./freebayes-1.3.4 --dont-left-align-indels -f ref_fasta.fa --region chr1 -@ 1000GP_chr1_sites_biallelic.vcf.gz input.bam | bcftools view -Ob -o out_freebayes_v1.3.4_call.bcf` 
Just be careful here, as freebayes encodes likelihoods in the GL field, so you need to use GLIMPSE using the --input-GL option.

Or you can run GATK as well using (please double-check this):
`gatk --java-options "-Xmx16g" HaplotypeCaller -R ref_fasta.fa -I input.bam -O out_gatk_call.vcf.gz --all-site-pls --L 1000GP_chr1_sites_biallelic.vcf.gz --alleles 1000GP_chr1_sites_biallelic.vcf.gz --output-mode EMIT_ALL_ACTIVE_SITES --min-pruning 0`

If you still have troubles, please send me an email, I can help even more and provide tools if necessary.
Hope this helps,

Simone

**Comment by ivargr** [comment]
Thanks a lot for clarifying!

I've now successfully been able to run GLIMPSE on various VCFs with genotype likelihoods, and GLIMPSE seems to work really well.

As I wrote in the other issue, this thing about resolution ended up being a misunderstanding from my side, as I didn't get good results at first and thought resolution might play a role, but the cause was other errors on my side.

Thanks, again.
